services/src/main/java/org/keycloak/broker/saml/mappers/UserAttributeMapper.java

```diff
@@ -99,61 +101,84 @@ public String getDisplayType() {
             } else {
-                context.setUserAttribute(attribute, value);
+                context.setUserAttribute(attribute, attributeToValues.get(attributeName));
             }
         }
     }
 
-    protected String getAttribute(IdentityProviderMapperModel mapperModel, BrokeredIdentityContext context) {
+    private String getAttributeName(IdentityProviderMapperModel mapperModel) {
+        String attributeName = mapperModel.getConfig().get(ATTRIBUTE_NAME);
+        if (attributeName == null) {
+            attributeName = mapperModel.getConfig().get(ATTRIBUTE_FRIENDLY_NAME);
+        }
+        return attributeName;
+    }
+
+    private void setIfNotEmpty(Consumer<String> consumer, List<String> values){
+        if (values != null && !values.isEmpty()) {
+            consumer.accept(values.get(0));
+        }
+    }
+
+    private Map<String, List<String>> getAttributeValues(IdentityProviderMapperModel mapperModel, BrokeredIdentityContext context) {
+        Map<String, List<String>> attributeToValues = new HashMap<>();
         String name = mapperModel.getConfig().get(ATTRIBUTE_NAME);
         if (name != null && name.trim().equals("")) name = null;
         String friendly = mapperModel.getConfig().get(ATTRIBUTE_FRIENDLY_NAME);
         if (friendly != null && friendly.trim().equals("")) friendly = null;
-        AssertionType assertion = (AssertionType)context.getContextData().get(SAMLEndpoint.SAML_ASSERTION);
+        AssertionType assertion = (AssertionType) context.getContextData().get(SAMLEndpoint.SAML_ASSERTION);
         for (AttributeStatementType statement : assertion.getAttributeStatements()) {
             for (AttributeStatementType.ASTChoiceType choice : statement.getAttributes()) {
                 AttributeType attr = choice.getAttribute();
                 if (name != null && !name.equals(attr.getName())) continue;
                 if (friendly != null && !friendly.equals(attr.getFriendlyName())) continue;
 
-                List<Object> attributeValue = attr.getAttributeValue();
-                if (attributeValue == null || attributeValue.isEmpty()) return null;
-                return attributeValue.get(0).toString();
+                List<String> attributeValue = attr.getAttributeValue().stream()
+                        .filter(Objects::nonNull)
+                        .map(Object::toString)
+                        .collect(Collectors.toList());
+                if (name != null) {
+                    attributeToValues.put(name, attributeValue);
+                } else if (friendly != null) {
+                    attributeToValues.put(friendly, attributeValue);
+                }
             }
         }
-        return null;
+        return attributeToValues;
     }
 
     @Override
     public void updateBrokeredUser(KeycloakSession session, RealmModel realm, UserModel user, IdentityProviderMapperModel mapperModel, BrokeredIdentityContext context) {
```

Comment:
I got a bit confused about how to test org/keycloak/broker/provider/IdentityProviderMapper.updateBrokeredUser

What kind of code do I need in the test so that 

`AssertionType assertion = (AssertionType) context.getContextData().get(SAMLEndpoint.SAML_ASSERTION);`

returns the new values for the attribute? 

```java
authenticate(getIdentityProviderModel(), "test-user-4", "new@email.com", true);

//Update SAML assertion i.e. add a value to the mobile attribute how do I do this?

brokerServerRule.stopSession(this.session, true);
        
this.session = brokerServerRule.startSession();

UserModel user = authenticate(getIdentityProviderModel(), "test-user-4", "new@email.com", false);

 assertThat(user.getAttribute("mobile"), containsInAnyOrder("555-666", "666-777"));

```